java.lang.ClassCastException: spoon.support.reflect.declaration.CtAnonymousExecutableImpl cannot be cast to spoon.reflect.declaration.CtConstructor

Me, again.

Using 5.2.0, I got a:
```
Caused by: java.lang.ClassCastException: spoon.support.reflect.declaration.CtAnonymousExecutableImpl cannot be cast to spoon.reflect.declaration.CtConstructor
	at spoon.reflect.factory.ExecutableFactory.createReference(ExecutableFactory.java:109)
	at spoon.support.reflect.declaration.CtExecutableImpl.getReference(CtExecutableImpl.java:145)
```

when parsing the following code:

```
class Foo {
	static {
		System.out.println("fooo");
	}
}
```

A test case to reproduce the exception:

```
public void testSpoon() {
		final spoon.Launcher launcher = new spoon.Launcher();
		launcher.addInputResource("src/test/resources/java/Foo.java");
		launcher.getEnvironment().setNoClasspath(true);
		launcher.getEnvironment().setComplianceLevel(8);
		launcher.buildModel();

		launcher.getModel().getElements(new TypeFilter<CtExecutable<?>>(CtExecutable.class) {
			@Override
			public boolean matches(final CtExecutable<?> exec) {
				try {
					System.out.println(exec.getReference());
				}catch(ClassCastException ex) {
					fail();
				}

				return super.matches(exec);
			}
		});
	}
```

--- a/src/main/java/spoon/reflect/factory/ExecutableFactory.java
+++ b/src/main/java/spoon/reflect/factory/ExecutableFactory.java
@@ -105,6 +105,8 @@ public class ExecutableFactory extends SubFactory {
 			return createReference(((CtMethod<T>) e).getDeclaringType().getReference(), ((CtMethod<T>) e).getType().clone(), e.getSimpleName(), refs);
 		} else if (e instanceof CtLambda) {
 			return createReference(e.getParent(CtType.class).getReference(), e.getType(), e.getSimpleName(), refs);
+		} else if (e instanceof CtAnonymousExecutable) {
+			return createReference(((CtAnonymousExecutable) e).getDeclaringType().getReference(), e.getType().clone(), e.getSimpleName());
 		}
 		return createReference(((CtConstructor<T>) e).getDeclaringType().getReference(), ((CtConstructor<T>) e).getType().clone(), CtExecutableReference.CONSTRUCTOR_NAME, refs);
 	}

--- a/src/test/java/spoon/test/reference/ExecutableReferenceTest.java
+++ b/src/test/java/spoon/test/reference/ExecutableReferenceTest.java
@@ -4,6 +4,7 @@ import org.junit.Test;
 import spoon.Launcher;
 import spoon.reflect.code.CtInvocation;
 import spoon.reflect.declaration.CtClass;
+import spoon.reflect.declaration.CtExecutable;
 import spoon.reflect.declaration.CtInterface;
 import spoon.reflect.declaration.CtMethod;
 import spoon.reflect.reference.CtExecutableReference;
@@ -124,5 +125,26 @@ public class ExecutableReferenceTest {
 		final List<CtExecutableReference<?>> kuuExecutables = kuu.getAllExecutables().stream().collect(Collectors.toList());
 		assertEquals(1, kuuExecutables.size());
 		assertEquals(kuu.getMethod("m").getReference(), kuuExecutables.get(0));
+	}
+
+	@Test
+	public void testCreateReferenceForAnonymousExecutable() {
+		final spoon.Launcher launcher = new spoon.Launcher();
+		launcher.addInputResource("src/test/resources/noclasspath/Foo4.java");
+		launcher.getEnvironment().setNoClasspath(true);
+		launcher.getEnvironment().setComplianceLevel(8);
+		launcher.buildModel();
+
+		launcher.getModel().getElements(new TypeFilter<CtExecutable<?>>(CtExecutable.class) {
+			@Override
+			public boolean matches(final CtExecutable<?> exec) {
+				try {
+					exec.getReference();
+				} catch (ClassCastException ex) {
+					fail(ex.getMessage());
+				}
+				return super.matches(exec);
+			}
+		});
 	}
 }

--- a/src/test/resources/noclasspath/Foo4.java
+++ b/src/test/resources/noclasspath/Foo4.java
@@ -1,0 +1,5 @@
+class Foo4 {
+	static {
+		System.out.println("fooo");
+	}
+}

